content/asp.net/getting-started/mvc.md

```diff
@@ -1,20 +1,122 @@
+The differentiator between a stock ASP.NET application and an MVC-based app are the services and facilities that MVC Core provides. In order to make these services available to the runtime and to our own services we need to first configure them as our application initializes. Here is the code for the ``ConfigureServices`` method in ``startup.cs`` which was included as part of the project template:
+
+```c#
+    public void ConfigureServices(IServiceCollection services)
+    {
+        // Add framework services.
+        services.AddMvc();
+    }
+```
+
+``AddMvc`` is an extension method of ``IServiceCollection`` that gives MVC access to the service container and configures it for use during request processing.  Next, we have the ``Configure`` method, which is the appropriate place to add our middleware:
+
+```c#
+    public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+    {
+        loggerFactory.AddConsole(Configuration.GetSection("Logging"));
+        loggerFactory.AddDebug();
+
+        if (env.IsDevelopment())
+        {
+            app.UseDeveloperExceptionPage();
+            app.UseBrowserLink();
+        }
+        else
+        {
+            app.UseExceptionHandler("/Home/Error");
+        }
+
+        app.UseStaticFiles();
+
+        app.UseMvc(routes =>
+        {
+            routes.MapRoute(
+                name: "default",
+                template: "{controller=Home}/{action=Index}/{id?}");
+        });
+    }
+
+```
+
+In the first part of this block you'll find some code that sets up the logging middleware, exception handling and finally the static files [middleware](static-files.md). Wrapping up the method is the call to ``UseMvc``, an extention method to ``IApplicationBuilder`` that configures the MVC middleware. MVC should be the last thing you provide, as it incurs the cost of processing a request and would add latency to things like static files. Keep this in mind as you order your middleware, and make sure that ``UseStaticFiles`` appears before the MVC middleware.  
+
+## Project Structure and Conventions
+In our solution created from the default project template there are a few points to note related to the folder layout and the conventions of the framework. In the root of the project we find some key classes, like the ``program.cs`` file with our application entry point, and the ``startup.cs`` class we were discussing above. There is a ``project_readme.html`` with information about the libraries used in the template and some helpful links related to MVC development.  
+
+You will find the familiar ``project.json`` file, some default application settings in the ``appsettings.json`` file, as well configuration hints for bundling and minification.  To enable our application to run on [Kestrel behind IIS Express](https://docs.asp.net/en/latest/fundamentals/servers.html#servers-and-commands), there is a ``web.config`` file with settings for our [host](hosts.md).
```

Comment:
That docs.asp.net page is very outdated, it hasn't been updated since RC1. Not sure if it's a good idea to link to it.